serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/States.kt

```diff
@@ -461,9 +464,19 @@ class MatrixChatRoom(private val msession: MatrixSession, val room_ids: List<Str
             got_messages
         }
     }
+    fun getDisplayNameForUser(sender: String) : String {
+        val (displayname, _) = msession.getDiplayNameAndAvatarFilePath(sender, room_id)
+        return displayname ?: ""
+    }
+    fun getUnformattedBody(formatted_body: String) : String {
+        val link_regex = Regex("<a href=\"https://matrix.to/#/[^:]*:[^>]*\">(.*)</a>")
```

Comment:
This is great, but the Regex value should be pulled out of the function so it doesn't get re-created all the time